core/server/services/routing/controllers/entry.js

```diff
@@ -63,12 +63,15 @@ module.exports = function entryController(req, res, next) {
              * @NOTE:
              *
              * Ensure we redirect to the correct post url including subdirectory.
+             *
+             * @NOTE:
+             * This file is used for v0.1 and v2. v0.1 returns relative urls, v2 returns absolute urls.
              */
-            if (post.url !== req.path) {
+            if (urlService.utils.absoluteToRelative(post.url, {withoutSubdirectory: true}) !== req.path) {
```

Comment:
Was curious why this would work for v0.1 with relative paths. This unit test makes it clear that the service also works with relative URLs:
```javscript
        it('extracts subdirectory from relative path', function () {
            configUtils.set('url', 'http://myblog.com/blog');

            urlService.utils.absoluteToRelative('/blog/test/', {withoutSubdirectory: true})
                .should.eql('/test/');
        });
```
Think this solution is good enough :+1: 

Only one comment regarding the note above:
```
             * @NOTE:
             *
             * Ensure we redirect to the correct post url including subdirectory.
```
When looking at it and then checking the use of `absoluteToRelative(post.url, {withoutSubdirectory: true})` makes it confusing. Maybe would be worth moving it to L:74 as that's what it corresponds to :)